blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/Containers.java

```diff
@@ -15,7 +15,133 @@
+                final int end = limit < 0 ? -1 : (start + limit);
+                Iterator<? extends T> iterator;
+                private T next;
+                int index = 0;
+                int iterableIndex = -1;
+                {
+                    // primer
+                    nextIterator();
+                    moveNext();
+                }
+                private void nextIterator() {
+                    iterableIndex++;
+                    if (iterableIndex < iterables.length) {
+                        iterator = iterables[iterableIndex].iterator();
+                    } else {
+                        iterator = null;
+                    }
+                }
+                private void moveNext() {
+                    // Skip until we hit our starting index
+                    while (index < start) {
+                        if (iterator == null) {
+                            return;
+                        }
+                        if (iterator.hasNext()) {
+                            iterator.next();
+                            index++;
+                        } else {
+                            nextIterator();
+                        }
+                    }
+
+                    // keep adding until we hit the limit or run out of items
+                    if (index < end || end < 0) {
+                        while (iterator != null && !iterator.hasNext()) {
+                            nextIterator();
+                            if (iterator == null) {
+                                return;
+                            }
+                        }
+
+                        next = iterator.next();
+                        index++;
+                    } else {
+                        iterableIndex = iterables.length;
+                        iterator = null;
+                    }
+                }
+
+                @Override
+                public boolean hasNext() {
+                    return iterator != null;
+                }
+
+                @Override
+                public T next() {
+                    try {
+                        return next;
+                    } finally {
+                        moveNext();
+                    }
+                }
+
+                @Override
+                public void remove() {
+                    throw new RuntimeException("remove() not implemented.");
+                }
+            };
+        }
+    }
+
+    /**
+     * Simple method to provide a combination of {@link Iterable} with the ability to implement a specific
+     * {@link #get(String)} method in a subclass. This will also iterate as efficiently as possible
+     * through the provided iterators to provide paged data - e.g. requesting page 0-9 will only
+     * iterate through the first 10 items, in the order of the provided {@link Iterable}
+     */
+    public abstract static class IterableContainer<T extends Resource> extends Container<T> {
```

Comment:
@cliffmeyers Ah ok. Sorry, I remember now, @kzantow mentioned this to me earlier. I guess my objection is only about reliability of this approach. Maybe its fine, given the fact that expectedBuildNumber and run id within a pipeline are always going to be unique.